src/desktop/src/index.js

```diff
@@ -21,84 +21,96 @@ import Index from 'ui/Index';
-            const data = hasDataToMigrate ? oldPersistedData : persistedDataFromRealm;
+    const bugsnagClient = bugsnag({
+        apiKey: 'fakeAPIkey',
+        appVersion: settings.version,
+        interactionBreadcrumbsEnabled: false,
+        collectUserIp: false,
+        user: { id: Electron.getUuid() },
+    });
+    bugsnagClient.use(bugsnagReact, React);
 
-            // Change provider on global iota instance
-            const node = get(data, 'settings.node');
-            changeIotaNode(node);
+    const ErrorBoundary = bugsnagClient.getPlugin('react');
 
-            // Update store with persisted state
+    if (Electron.mode === 'tray') {
+        Electron.onEvent('store.update', (payload) => {
+            const data = JSON.parse(payload);
             store.dispatch(mapStorageToStateAction(data));
+        });
+    } else {
+        initialiseStorage(getEncryptionKey)
+            .then(() => {
+                const oldPersistedData = Electron.getAllStorage();
+                const hasDataToMigrate = !isEmpty(oldPersistedData);
+
+                if (hasDataToMigrate) {
+                    Object.assign(oldPersistedData.settings, {
+                        completedMigration: false,
+                    });
+                }
+
+                // Get persisted data from Realm storage
+                const persistedDataFromRealm = mapStorageToState();
+                const data = hasDataToMigrate ? oldPersistedData : persistedDataFromRealm;
+
+                // Change provider on global iota instance
+                const node = get(data, 'settings.node');
+                changeIotaNode(node);
 
-            // Assign accountIndex to every account in accountInfo if it is not assigned already
-            store.dispatch(assignAccountIndexIfNecessary(get(data, 'accounts.accountInfo')));
+                // Update store with persisted state
+                store.dispatch(mapStorageToStateAction(data));
 
-            // Proxy realm changes to Tray application
-            realm.addListener('change', () => {
-                const data = mapStorageToState();
-                Electron.storeUpdate(JSON.stringify(data));
+                // Assign accountIndex to every account in accountInfo if it is not assigned already
+                store.dispatch(assignAccountIndexIfNecessary(get(data, 'accounts.accountInfo')));
+
+                // Proxy realm changes to Tray application
+                realm.addListener('change', () => {
+                    const data = mapStorageToState();
+                    Electron.storeUpdate(JSON.stringify(data));
+                });
+
+                // Start Tray application if enabled in settings
+                const isTrayEnabled = get(data, 'settings.isTrayEnabled');
+                Electron.setTray(isTrayEnabled);
+
+                // Show Wallet window after inital store update
+                Electron.focus();
+            })
+
+            .catch((err) => {
```

Comment:
Are we able to send the error to Bugsnag as well?

Reply:
Added